### PR TITLE
Add more Prometheus metrics

### PR DESCRIPTION
**Pull request by sfan5**
Inspired by the [patches](https://github.com/pandorabox-io/minetest_docker/tree/master/patches) pandorabox uses I decided to make the existing [prometheus](https://prometheus.io/) support more useful by adding some metrics that actually matter.

Note: This renames `minetest_core_map_save_time` to `minetest_map_save_time` and changes the unit to microseconds.

## To do

This PR is a Ready for Review.

## How to test

Build Minetest with prometheus-cpp and hook it up to a prometheus instance or use `curl http://localhost:30000/metrics`

### CODE REVIEW

**Comment by wsor4035** [comment]
it would be nice to see https://github.com/minetest/minetest/pull/10980 supported at some point. probably outside of the scope of this PR

File: src/map.cpp [hunk 1]
```diff
@@ -1218,7 +1226,12 @@ ServerMap::ServerMap(const std::string &savedir, IGameDef *gamedef,
 	m_savedir = savedir;
 	m_map_saving_enabled = false;
 
-	m_save_time_counter = mb->addCounter("minetest_core_map_save_time", "Map save time (in nanoseconds)");
+	m_save_time_counter = mb->addCounter(
+		"minetest_map_save_time", "Time spent saving blocks (in microseconds)");
```
Comment (rubenwardy) [inline src/map.cpp:1230]: Should probably use a consistent naming pattern: `minetest_core_`. Same applies to other added metrics.

The reason to prefix by `minetest_core_` is to distinguish between engine metrics and mod metrics (which may be added by another prometheus service or by a future API)

Comment (rubenwardy) [reply src/map.cpp:1230]: > Note: This renames minetest_core_map_save_time to minetest_map_save_time and changes the unit to microseconds.

Ah...   Idk then

Comment (sfan5) [reply src/map.cpp:1230]: The naming scheme here is: minetest_&lt;area>_&lt;thing>
where &lt;area> is one of core (server.cpp), env (serverenvironment.cpp), map (map.cpp), emerge (emerge*.cpp)

When mod metrics are added I think it's fine to let mods name their stuff whatever they want but give advice to use `minetest_mod_<modname>_<thing>`

Comment (rubenwardy) [reply src/map.cpp:1230]: OK fair enough

**Review by rubenwardy — APPROVED** [review]

**Comment by rubenwardy** [comment]
Haven't tested

**Pull request merged**